### PR TITLE
[PORT] That PR where I port the PR that refactors the entire preferences system

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
Port of https://github.com/BeeStation/BeeStation-Hornet/pull/7016

## Why It's Good For The Game
Removes savefile jank

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Honestly I didn't have time to set up a database yet, but was told this was wanted for tomorrow, so good luck

</details>

## Changelog
:cl: AffectedArc07
refactor: Converted all character + preference data to live in the database instead of savefiles
server: You will need to convert all your old savefiles. Good luck.
tweak: Changed preferred squad and preferred pilot role to be linked to the active character. Your squad and pilot role have likely been reset to defaults.
code: Put the widescreen preference in toggles2 instead of it being standalone
/:cl:

### CODE REVIEW

**Comment by Kmc2000** [comment]
based

File: SQL/database_changelog.txt [hunk 1]
```diff
@@ -2,12 +2,66 @@ Any time you make a change to the schema files, remember to increment the databa
...
+	`uplink_loc` VARCHAR(64) NOT NULL COLLATE 'utf8mb4_general_ci',
+	`features` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_general_ci',
+	`custom_names` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_general_ci',
+	`helmet_style` VARCHAR(64) NOT NULL COLLATE 'utf8mb4_general_ci',
+	`preferred_ai_core_display` VARCHAR(64) NOT NULL COLLATE 'utf8mb4_general_ci',
+	`preferred_security_department` VARCHAR(32) NOT NULL COLLATE 'utf8mb4_general_ci',
+	`joblessrole` TINYINT(4) UNSIGNED NOT NULL,
+	`job_preferences` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_general_ci',
+	`all_quirks` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_general_ci',
+	`equipped_gear` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_general_ci',
```
Comment (Crossedfall) [inline SQL/database_changelog.txt:54]:
```suggestion
	`equipped_gear` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_general_ci',
	`preferred_squad` VARCHAR(32) NOT NULL COLLATE 'utf8mb4_general_ci',
	`preferred_pilot_role` VARCHAR(32) NOT NULL COLLATE 'utf8mb4_general_ci',
```

**Review by Bokkiewokkie — DISMISSED** [review]

**Review by Bokkiewokkie — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -186,3 +186,6 @@ Temporary Items
 
 # Common build tooling, C B T
 !/tools/build
+
+# Ignore savefiles
+tools/savefileimporter/player_saves/
```
Comment (Crossedfall) [inline .gitignore:191]:
```suggestion
```

This should go in the savefileimport's gitignore.

File: tools/savefileimporter/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+config.json
+rust_g.dll
```
Comment (Crossedfall) [inline tools/savefileimporter/.gitignore:2]:
```suggestion
rust_g.dll
player_saves
```

**Review by Bokkiewokkie — APPROVED** [review]

**Comment by Bokkiewokkie** [comment]
If something breaks we can fix it in post

**Pull request merged**